cmd/explaintest/r/new_character_set.result

```diff
@@ -0,0 +1,42 @@
+drop table if exists t;
+set names utf8mb4;
+create table t (a varchar(255) charset utf8mb4);
+insert into t values ('一二三四一');
+select hex(a) from t;
+hex(a)
+E4B880E4BA8CE4B889E59B9BE4B880
+select a from t;
+a
+一二三四一
+set @@character_set_results = 'gbk';
+select a from t;
+a
+һ������һ
+drop table if exists 一;
+create table 一 (二 char(20));
+show create table 一;
+Table	Create Table
+һ	CREATE TABLE `һ` (
+  `��` char(20) DEFAULT NULL
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
+drop table if exists t;
+set names utf8mb4;
+create table t (a varchar(255) charset gbk, b varchar(255) charset utf8mb4);
+insert into t values ('一', '一');
+set @@character_set_results = null;
+select * from t;
+a	b
+һ	一
+set @@character_set_results = BINARY;
+select * from t;
+0x61	0x62
+0xD2BB	0xE4B880
```

Comment:
I tried it in MySQL 8.0.25 and here's what happened:
```
mysql> set @@character_set_results = BINARY;
Query OK, 0 rows affected (0.00 sec)

mysql> select * from t;
+------+------+
| a    | b    |
+------+------+
| һ    | 一   |
+------+------+
1 row in set (0.00 sec)
```

Reply:
`set @@character_set_results = BINARY` means perform no conversion of result sets in MySQL, it should be same with `set @@character_set_results = NULL`